TypeScript v6: Property 'language' does not exist on type 'never'

### What version of `starlight` are you using?

0.38.3

### What version of `astro` are you using?

6.1.6

### What package manager are you using?

pnpm

### What operating system are you using?

mac

### What browser are you using?

N/A

### Describe the Bug

After updating TypeScript to v6, I got the following error when running `tsc --noEmit`:


```
> tsc --noEmit

node_modules/.pnpm/@astrojs+starlight@0.38.3_astro@6.1.6_@types+node@24.12.2_rollup@4.60.1_typescript@6.0.2_/node_modules/@astrojs/starlight/utils/i18n.ts:196:38 - error TS2339: Property 'language' does not exist on type 'never'.

196  return wellKnownRTL.includes(locale.language) ? 'rtl' : 'ltr';
                                         ~~~~~~~~


Found 1 error in node_modules/.pnpm/@astrojs+starlight@0.38.3_astro@6.1.6_@types+node@24.12.2_rollup@4.60.1_typescript@6.0.2_/node_modules/@astrojs/starlight/utils/i18n.ts:196
```

It seems that this is caused by the fact that TypeScript v6 has a newer DOM library; thus, `locale` becomes `never` in [this line](https://github.com/withastro/starlight/blob/db10448ef5faa8c00a5474d5d572c9e00f667708/packages/starlight/utils/i18n.ts#L196) after the two `if` statements above.

For a short-term workaround, we can just add a `@ts-ignore` comment for this line; In the long term, https://github.com/withastro/starlight/pull/3572 should be the correct path to avoid such issues.

### Link to Minimal Reproducible Example

https://github.com/issueset/starlight-ts-6-issue.git

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/yellow-stingrays-report.md
+++ b/.changeset/yellow-stingrays-report.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Fixes a type-checking issue for users on newer versions of TypeScript

--- a/packages/starlight/utils/i18n.ts
+++ b/packages/starlight/utils/i18n.ts
@@ -193,6 +193,9 @@ function getLocaleDir(locale: Intl.Locale): 'ltr' | 'rtl' {
 	}
 	// Firefox does not support `textInfo` or `getTextInfo` yet so we fallback to a well-known list
 	// of right-to-left languages.
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore — This is a type error with newer versions of TypeScript’s DOM types.
+	// TODO: remove or switch to @ts-expect-error in #3572
 	return wellKnownRTL.includes(locale.language) ? 'rtl' : 'ltr';
 }
 

